### ISSUE TITLE
Kindly review.

### ISSUE THREAD

**Issue by ventuno** [ventuno/bf#1]
Dear @alekseykulikov, I just completed the exercise [1].
The instructions on how to run the project are on the README.md [2] file, it should be pretty straightforward as it's an npm project that uses gulp to build the source code.

A few notes:
1) I spent about 1.5/2 full days on the project;
2) Whenever I do these kind of tests, I always like to learn something new, this time I wanted to play around with gulp and had a lot of fun!
3) I didn't have the "edit" icon to change the smiley, I used a placeholder (the butterfly), I hope that's not a big deal. If you want you can send me the correct one, it should be as easy as placing it in src/components/MoodBox/assets/EditMood.svg.
4) I did all the measurements on the png file on Preview, so if the design looks a little off it's because that's the best I could do without Sketch. In general, most of the margins/colors/sizes are controlled by Less variables. Changing them should be as simple as updating those variables. That's actually a pretty common scenario at my work, I implement as per specifications and sometimes sit with a designer to achieve desired pixel-perfection.
5) I have an alternative implementation for the rating bar, it uses percentages only and scales way better on mobile.
6) I might be doing some changes to the build part tomorrow, as I wanted to add a linter to the build. I hope that's alright.

[1] https://github.com/ventuno/bf
[2] https://github.com/ventuno/bf/blob/master/README.md

**Comment by alekseykulikov** [comment]
Hello Ernesto!

I really like aesthetic you put to this project: from name and readme to code and execution.

I have a few notes for you (if you have time you can try to tackle them and learn something new):
- Do you familiar with ES6? All our code is written on ES6. I see you're already using babel, so it won't be a problem to add a few presets. Our code and babel config:
![screen shot 2016-08-29 at 10 27 19](https://cloud.githubusercontent.com/assets/158189/18045872/65455444-6dd5-11e6-9220-abe2f27bff1c.png)
![screen shot 2016-08-29 at 10 28 34](https://cloud.githubusercontent.com/assets/158189/18045871/6540ca00-6dd5-11e6-8f4c-bc9a75f94f0b.png)

- Do you familiar with [BEM or SUITCSS](https://medium.com/@jeremejevs/react-suit-css-429cc4269885#.ep1z9yu8u)? I see your css could benefit from some naming convention. Our css:
![screen shot 2016-08-29 at 10 34 13](https://cloud.githubusercontent.com/assets/158189/18045885/6c0382d8-6dd5-11e6-8f21-d583c2d8fc70.png)

- I see you set up eslint, but gulpfile.js has issues. Why do you prefer custom config instead of [airbnb](https://github.com/airbnb/javascript/tree/master/packages/eslint-config-airbnb-base) or [standard](standardjs.com) presets? Also it's a good practice to run linter as a part of `npm test`
![screen shot 2016-08-29 at 10 27 44](https://cloud.githubusercontent.com/assets/158189/18045893/725a05bc-6dd5-11e6-8f2e-4ae741ab3079.png)

- Why did you put `server.js` to `/src`? Also you need to clean dependencies/devDependencies. `react` and `react-dom` are just dependencies, and devDependencies are missing `eslint`.

I think you did a great job for this short period of time.
If you curious you can check our current execution: https://butterfly.ai/demo-question?v=4

Best,
Aleksey

**Comment by ventuno** [comment]
Thanks a lot @alekseykulikov for the feedback!

I spent some time to implement some of the recommendations:
1) I converted all components to ES6 classes, I believe I still have a lot to learn about ES6 as I don't use it every day (yet), but I'm very happy I had the opportunity to give it a try.
2) I really like the way SUITCSS organizes the code, but I prefer the way the code in the article implements it (e.g.: extending `React.Component` and implementing the `cn` method), I'll try to spend some time implementing it that way.
3) You're right on linting the `gulpfile` as well! I don't have the linter configured in my editor and run it manually on the console. I added it to the `eslint` task and it should be fine now :-).
I took the basic rules from the `react` source code itself. I know Airbnb set the standard for React dev style, so I'll definitely look into that separately.
4) I understand you'd probably prefer the `server.js` in the root directory of the project and I kind of disagree, I like to have anything that is actual code in `src/`. In a more real life scenario, I'd probably create a `client/` and `server/` sub directory.
